File: demo1/src/main/java/com/ni/TextDeal.java
package com.ni;

import java.io.*;
import java.util.Arrays;
import java.util.HashSet;

public class TextDeal {
    public static void main(String[] args) throws IOException {
        File file = new File("D:\\IdeaProjects\\java_flink\\demo1\\src\\main\\resources\\人群标签.txt");
        FileReader fileReader = new FileReader(file);
        HashSet<String> tags = new HashSet<>();
        try (BufferedReader tagsReader = new BufferedReader(fileReader)) {
            String line;
            while ((line = tagsReader.readLine()) != null) {
                tags.addAll(Arrays.asList(line.split("\\|")[1].split(",")));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        for (String tag : tags) {
            System.out.print(tag + ",");
        }

    }
}

File: demo1/src/main/scala/TextDeal.scala
import java.io.{ BufferedReader, FileReader }
import scala.collection.mutable

/**
 * program: java_flink
 * description:
 * author: Ni
 * create: 2021-11-17 15:00
 * */
object TextDeal {
    def main(args: Array[String]): Unit = {
        val tagFileReader = new FileReader("D:\\IdeaProjects\\java_flink\\demo1\\src\\main\\resources\\人群标签.txt")
        val tagsBufferReader = new BufferedReader(tagFileReader)
        var line = ""
        val tags = new mutable.HashSet[String]()

        //while(true)  {
        //    println(line)
        //    val tagsInLine =
        //        line
        //            .split("\\|")(1)
        //            .split(",")
        //    tagsInLine.foreach(tags.add)
        //}
        println(tags.mkString(","))
    }
}
